src/devices/Charlieplex/CharlieplexSegment.cs

```diff
@@ -210,5 +215,88 @@ public void Dispose()
                 _gpioController = null!;
             }
         }
+
+        // IOutputSegment Implementation
+        // Only supported when shift register is connected with GPIO
+
+        /// <summary>
+        /// The length of the segment; the number of GPIO pins it exposes.
+        /// </summary>
+        public int Length => _segment.Length;
+
+        /// <summary>
+        /// Segment values.
+        /// </summary>
+        PinValue IOutputSegment.this[int index]
```

Comment:
Sorry I only added a comment on Length property, but I really meant on all the methods of the interface. Looks like you implemented all of them through the interface, where we really want them to be implemented through the class so that they can be called without casting.

Reply:
Ya, that is intentional. These methods and then ones on the class are not intended to be used together. As implemented, they won't play nicely in all cases.  This is a layer on top, which is why I went with the interface methods like that. I could have made a new class for each one, but that seemed like overkill.